engine/api/cache.go

```diff
@@ -0,0 +1,61 @@
+package api
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/gorilla/mux"
+
+	"github.com/ovh/cds/engine/api/objectstore"
+	"github.com/ovh/cds/sdk"
+)
+
+func (api *API) postPushCacheHandler() Handler {
+	return func(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+		vars := mux.Vars(r)
+		projectKey := vars["key"]
+		tag := vars["tag"]
```

Comment:
tag must checked with a pattern (to avoid url encoding issues with swift)

Reply:
done inside command

Reply:
should be done here too